### PR TITLE
2596 - launchdarkly clean up and basic logic

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:* /bcgov/entity#2596

*Description of changes:*
- removed old library ld-vue
- updated to use launchdarkly-js-client-sdk (same as create ui)
- added FeatureFlags class
- added basic logic to block the app from loading up if it is an NR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: public/config/configuration.json [hunk 1]
```diff
@@ -7,5 +7,5 @@
   "PAY_API_URL": "https://pay-api-dev.pathfinder.gov.bc.ca/api/v1/",
   "KEYCLOAK_CONFIG_PATH": "/local-keycloak-config-url/keycloak.json",
   "ADDRESS_COMPLETE_KEY": "RH99-RN99-GB72-FW86",
-  "LD_CLIENT_ID": "5dc1c5fd1d213d087f9e1024"
+  "LD_CLIENT_ID": "5e5d5af4574f7e09ab53d03b"
```
Comment (ochiu) [inline public/config/configuration.json:10]: Discussed with @lekshmimallika-aot and we think we probably should use a single LD_CLIENT_ID so we can disable the features in one location. Thoughts?

File: src/utils/fetch-config.ts [hunk 2]
```diff
@@ -68,6 +69,9 @@ export async function fetchConfig (): Promise<void> {
   window['ldClientId'] = ldClientId
   console.info('Set Launch Darkly Client ID.')
 
+  // initialize Launch Darkly
+  await initLDClient()
+
```
Comment (severinbeauvais) [inline src/utils/fetch-config.ts:74]: I'd prefer to see this init visible in `main.ts` (same as in Create UI).

File: src/views/Dashboard.vue [hunk 3]
```diff
@@ -293,7 +290,8 @@ export default {
 
   mounted (): void {
     // Launch Darkly flag usage example:
-    // console.log('coopsVersion =', this.flags.coopsVersion)
+    // First import common class FeatureFlags
+    // console.log('bcrs-create-ui-enabled =', featureFlags.getFlag('bcrs-create-ui-enabled'))
   },
```
Comment (severinbeauvais) [inline src/views/Dashboard.vue:295]: Looks like all of `mounted()` can be deleted :)

File: public/config/configuration.json (hunk 1 above)
Comment (severinbeauvais) [reply public/config/configuration.json:10]: Sure, but what about the different environments?

And if you change it here, do you also have to change the config map keys for Dev and/or Test and/or Prod?

File: src/utils/fetch-config.ts (hunk 2 above)
Comment (ochiu) [reply src/utils/fetch-config.ts:74]: The validation piece is right after this line of code and we need the ldClientId to be set for the `initLDClient()`

Should I move the feature flag check elsewhere?

File: public/config/configuration.json (hunk 1 above)
Comment (ochiu) [reply public/config/configuration.json:10]: I am not sure if we have different client ids per environment, @lekshmimallika-aot ?

But yes, if we change it here we need to change it for dev, test, and prod so it shares the same set of properties

File: src/utils/fetch-config.ts (hunk 2 above)
Comment (severinbeauvais) [reply src/utils/fetch-config.ts:74]: No, I had in mind the FF check here, but I forgot that you need to load the config before init (catch-22).

What do you think about moving this FF check to main.ts and wrapping it around the app start (same as Create UI)? You could then log a message if the flag is false (and/or throw an error that the existing catch logic would handle).

File: src/views/Dashboard.vue (hunk 3 above)
Comment (ochiu) [reply src/views/Dashboard.vue]: Done.

File: src/utils/fetch-config.ts (hunk 2 above)
Comment (ochiu) [reply src/utils/fetch-config.ts:74]: Sounds good, will do that

File: public/config/configuration.json (hunk 1 above)
Comment (lekshmimallika-aot) [reply public/config/configuration.json:10]: Different environments will have different client ID. That will have to be updated in the config map of the respective environments. Here we can keep this one

**Pull request merged**